src/OutputWriters/netcdf_output_writer.jl

```diff
@@ -395,7 +395,7 @@ function NetCDFOutputWriter(model, outputs; filename, schedule,
     if mode == "c"
         for (dim_name, dim_array) in dims
             defVar(dataset, dim_name, array_type(dim_array), (dim_name,),
-                   compression=compression, attrib=default_dimension_attributes[dim_name])
+                   deflatelevel=compression, attrib=default_dimension_attributes[dim_name])
```

Comment:
What is the motivation for using the name `compression` in the user interface for `NetCDFOutputWriter`, but assigning it to `deflatelevel`? Is this standard / intuitive? Or should we also change `compression` to `deflatelevel`.

Reply:
I believe we should change compression to `deflatelevel` and in the docstring refer to NCDatasets.jl docs

Reply:
Although `compression` is more intuitive to `deflatelevel`....

Reply:
IMO compression is way more intuitive than `deflatelevel`, so I vote for leaving it the way it is. Does JLD2 have a similar option? What keyword do they use?

Reply:
I don't think the JLD2OutputWriter has this option, but looking at the JLD2 package I found:

https://juliaio.github.io/JLD2.jl/dev/compression/

We could use `compress::Boolean = true` as a keyword for both output writers and for the NetCDFOutputWriter also have a `deflatelevel` kwarg with some default value which will be passed if `compress = true`?

Reply:
Hm...

```
Note that CodecZlib.jl must be explicitly installed for compression/decompression to work. If the package is not already loaded JLD2 will try to dynamically load it, but it is recommended to do it explicitly, e.g. using JLD2, CodecZlib.
```

OK, so not sure if we want to deal with the JLD2OutputWriter in this PR, but in anticipation of potentially enabling compression for JLD2OutputWriter I still vote for a `compress::Bool` kwarg and then an additional kwarg with a default value, e.g., `deflatelevel::Int=4`.

Reply:
I'm okay with this solution

Reply:
I think we should pass kwargs to NCDatasets without changing their name.

Reply:
OK, this is a very good point!